paddle/fluid/operators/math/blas.h

```diff
@@ -46,6 +46,17 @@ namespace paddle {
 namespace operators {
 namespace math {
 
+struct MatDim {
```

Comment:
Given that MatDim is only used in Blas::MatMul, it seems that we should move `MatDim` into `Blas::MatDim`?

Also, the motivation of the introduction of MatDim is pretty confusing -- why cannot we use DDim for representing the shape of a matrix?

Reply:
I changed the name to `MatDescriptor`.

Not only the dimension but also the layout and stride of a memory buffer is described. It is clearer to have an independent structure to describe (transpose, stride, batch_size, dimension) together instead of reusing DDim.